widget/entry.go

```diff
@@ -378,124 +395,131 @@ func (e *Entry) TypedKey(key *fyne.KeyEvent) {
 
 	switch key.Name {
 	case fyne.KeyBackspace:
-		e.RLock()
-		isEmpty := provider.len() == 0 || (e.CursorColumn == 0 && e.CursorRow == 0)
-		e.RUnlock()
+		isEmpty := e.GetField(func() interface{} {
+			return provider.len() == 0 || (e.CursorColumn == 0 && e.CursorRow == 0)
```

Comment:
Apparently I can't comment on lines that haven't changed, but on line 387, is it safe to get the textProvider without read lock?